lib/friendly_id/finder_methods.rb

```diff
@@ -31,7 +31,9 @@ def find(*args, allow_nil: false)
       first_by_friendly_id(id).tap { |result| return result unless result.nil? }
       return super(*args) if potential_primary_key?(id)
 
-      raise_not_found_exception(id) unless allow_nil
+      raise_not_found_exception(id)
+    rescue => e
+      allow_nil ? nil : raise(e)
```

Comment:
I know we talked about exception control flow and I'm definitely not a fan, however I really am not aware of another strategy to use here. I'm open to suggestions though.